flocker/control/test/test_clusterstate.py

```diff
@@ -156,17 +160,46 @@ def test_expiration(self):
         (in seconds) old are wiped.
         """
         service = self.service()
-        app_node = NodeState(hostname=u"10.0.0.1", uuid=uuid4(),
-                             manifestations=None, devices=None, paths=None,
-                             applications=[APP1], used_ports=[])
-        service.apply_changes([app_node])
-        self.clock.advance(EXPIRATION_TIME - 1)
+        service.apply_changes([self.WITH_APPS])
```

Comment:
`apply_changes_with_source` is now the recommended public API to use, and in fact what does get used. But only one of the tests uses it. I would suggest all the tests use it and maybe have one bonus test for the backwards compat layer.